src/DataHandling/Logs/DataDogAudit.php

```diff
@@ -165,22 +271,242 @@ protected function removeColumnIfOnlyUnchanged($diffArray)
 
     /**
      * Method for getting key storing information about given on input column name with association. Can be override for customization needs.
-     * @param \DataDog\AuditBundle\Entity\AuditLog $currentVersion
-     * @return string name 
+     *
+     * @param AuditLog $currentVersion
+     *
+     * @return string name
      */
     protected function getColumnNameForAssociation(AuditLog $currentVersion)
     {
-        return $currentVersion->getTarget()->getTbl();
+        $joinTableName = $currentVersion->getTbl();
+        if (isset($this->cache['assocTable'][$joinTableName])) {
+            return $this->cache['assocTable'][$joinTableName];
+        }
+
+        foreach ($this->getEntitiesClassMetaData()->getAssociationMappings() as $assMapping) {
+            if (isset($assMapping['joinTable'])) {
+                $assJoinTable = $assMapping['joinTable']['name'];
+                $this->cache['assocTable'][$assJoinTable] = $assMapping['fieldName'];
+                if ($joinTableName === $assJoinTable) {
+                    return $assMapping['fieldName'];
+                }
+            } elseif (is_null($currentVersion->getTarget())
+                && $currentVersion->getSource()->getClass() === $assMapping['targetEntity']
+            ) { // ManyToOne mapping
+                $this->cache['assocTable'][$joinTableName] = $assMapping['fieldName'];
+
+                return $assMapping['fieldName'];
```

Comment:
From Symfony coding standard "Do not use else, elseif, break after if and case conditions which return or throw something;"

Reply:
as I understand this rule:
```php
// bad
if ($a) {
    return 5;
} else {
    $b = ...
    ...
// good
if ($a) {
    return 5;
}
$b = ...
...
```

My case is like this:
```php
foreach($array as $el) {
    if ($e->aaa()) {
        return 1;
    }
    if {$e->bbb()) {
        return 2;
    }
}

// when nothing matched, throw error
throw new \Exception();
```

We probably have to discuss the style guide in a Cube Meeting.

Reply:
ok, but it applies for 291 line

Reply:
no, I do not use "else", "elseif", "break"
after
"return" or "throw" (which are in "if" or "case").

I do use "throw" after "if".

Reply:
but in line 291 you have return and in 293 elseif. They are on a different level, but it looks like this rule applies for it (correct me if I am wrong)

Reply:
I use
"throw"
afer
"elseif"

NOT
"elseif"
after
"throw"

Reply:
But I have written in last post about return statement, not throw statement.